object/service.proto

```diff
@@ -688,3 +706,19 @@ message GetRangeHashResponse {
   // transmission.
   neo.fs.v2.session.ResponseVerificationHeader verify_header = 3;
 }
+
+// Replicate RPC request
+message ReplicateRequest {
+  // Object to be replicated.
+  neo.fs.v2.object.Object object = 1;
+
+  // Signature of all other request fields serialized in Protocol Buffers v3
+  // format in ascending order of fields.
+  neo.fs.v2.refs.Signature signature = 2;
+}
+
+// Replicate RPC response
+message ReplicateResponse {
+  // Operation execution status with one of the enumerated codes.
+  neo.fs.v2.status.Status status = 1;
```

Comment:
I like the simplicity of it, but this breaks https://github.com/nspcc-dev/neofs-sdk-go/blob/master/client/response.go and associated code around it. Does it affect integration?

Reply:
nope, integration is ok cuz this is purely new functionality. If protocol doesnt require signature (if we keep the current status-only response), then applications must not too

the only tricky thing i see is https://github.com/nspcc-dev/neofs-api/issues/201#issuecomment-1891383454, but there are app-side workarounds

the main thing for us is to think carefully about whether we need a signature from the protocol pov. And ofc, when in doubt, it is better to add it and then make it optional, because further additions must be optional for the backward compat sake. But, at the same time, the benefits are questionable while the resource costs are real

im leaning more towards the option without a signature as more efficient, but I'm not completely sure yet. Share ur thoughts pls @roman-khimov @carpawell

Reply:
Signatures make some sense for insecure connections or for rerouted messages, but for TLS-enabled 1:1 calls they're irrelevant. And then there is also some benefit in having insecure unsigned option for fast communication over known good channels. So if this inconsistency is not an obstacle, we better go without signatures.

Now for the other data like epochs, is it important for us to get it with _every_ reply? I doubt that, but maybe I'm missing something. This data costs nothing to transmit/receive.

Reply:
same thoughts about transport security. Can we spread this to requests also? In proposed changes, replication requests are signed while responses are not

> to get it with _every_ reply

absolutely not. For me, this is redundancy useful for a narrow set of tasks (mostly system ones), which should not be unconditionally transferred. I see the behavior: if client set flag in the request, epoch/smth will be returned, not by default. Although the amount of data like epoch is insignificant, it is still pointless to transfer it when not needed by the client

Reply:
i think a signature absence is acceptable for a system-only RPC response